jahorwitz/slopopedia-api
# feat: added endpoint for getting image url & changed movie model

## Describe your changes

### Movie Model Changes
Removed photo as it is no longer needed given slop-42 solution (see that PR for more info).

Added two new fields:
1. imageKey - the unique key for the movie's image in the s3 bucket that is returned by Danielle's endpoint.
2. imageUrl - a virtual field that returns a presigned url for the image in the s3 bucket.

### Endpoint Changes
I created an endpoint for fetching images using a get request to `/api/movies/:s3Key`

This allows for us to grab images for movies easily without having to modify the s3 bucket policy

### Misc. Changes
Added error handling to s3 functions as they were crashing the server when you hit the endpoint without the correct info.

### Testing instructions
1. Make a PUT request to `api/movies` using the instructions provided in the video on the slop-63 PR to upload an image to s3.
3. copy `uniqueKey` from the response.
**TO TEST ENDPOINT:**
4. make a `GET` request to `/api/movies/uniqueKey` or navigate to it in your web browser.
**TO TEST MOVIE MODEL:**
5. Create a movie in the DB with the imageKey set to your `uniqueKey` value.
6. Check the link supplied by the imageUrl field.
## Issue ticket number and link
## Checklist before requesting a review

- [ ] If there are changes to custom resolvers, I have added / updated automated tests
